telemetry/telemetry.go

```diff
@@ -70,16 +80,16 @@ func runTelemetry(args []string) error {
 	}
 
 	var wg sync.WaitGroup
-	// serverControlSignal channel is a channel that will be used to notify gnmi server of sigcall which should stop server
-	var serverControlSignal = make(chan int, 1)
+	// serverControlSignal channel is a channel that will be used to notify gnmi server to start, stop, restart, depending of syscall or cert updates
+	var serverControlSignal = make(chan ServerControlValue, 1)
 	sigchannel := make(chan os.Signal, 1)
 	signal.Notify(sigchannel, syscall.SIGTERM, syscall.SIGQUIT)
 
-	wg.Add(1)
-	go startGNMIServer(telemetryCfg, cfg, serverControlSignal, &wg)
+	go signalHandler(serverControlSignal, sigchannel)
```

Comment:
Why do we remove wg from signalHandler?

Reply:
Initially, I thought it doesn't matter if signalHandler is in the wg, since if startGNMIServer is the critical goroutine, if that exits we can have the process exit, but I think it is fine to add it to the wg to make sure all goroutines exit cleanly, avoid any potential issues or leaks.

Reply:
Added back to wg